app/src/components/TypeBadgeIcon.js

```diff
@@ -0,0 +1,15 @@
+import React from "react";
+
+export const TypeBadgeIcon = ({ type }) => (
+  <span className={`inline-block whitespace-no-wrap rounded-full text-center h-8 w-8 bg-type-${type.toLowerCase()}`}>
+    <svg className="fill-current text-gray-900 inline align-text-bottom" width="14" viewBox="0 0 512 512">
+      <use xlinkHref={`./icons/types/${type.toLowerCase()}.svg#${type.toLowerCase()}`}></use>
```

Comment:
I'm not sure how this link is working with Next.js / Webpack. We have to check its working in a prod build

Reply:
ref would be better inlined in the doc yes

Reply:
I just checked, it should be an absolute `/icons/...` which it absolute to the `public` directory